pkg/kubelet/kubelet_node_status.go

```diff
@@ -509,11 +509,30 @@ func (kl *Kubelet) tryUpdateNodeStatus(tryNumber int) error {
 		}
 	}
 
+	areRequiredLabelsNotPresent := false
```

Comment:
I expected to see an e2e for kubelet that orchestrated this (node e2e) - roughly start kubelet, verify labels show up, then stop kubelet, change the labels on the node via the test to be a different OS, then verify they go back.  then do the same thing without restarting the kubelet for both labels